tests/Unit/Utilities/Test_Tuple.cpp

```diff
@@ -262,4 +263,9 @@ SPECTRE_TEST_CASE("Unit.Utilities.Tuple", "[Utilities][Unit]") {
     CHECK(tuple_head<2>(tuple) == std::tuple<int, float>{1, 2.});
     CHECK(tuple_tail<2>(tuple) == std::tuple<double, std::string>{3., ""});
   }
+  {
+    INFO("From typelist");
+    static_assert(std::is_same_v<tuple_from_typelist<tmpl::list<int, float>>,
```

Comment:
check where the same type occurs multiple times, you have only 1 element, and you have 0 elements